tests/mocks/MockDesktop.hh

```diff
@@ -0,0 +1,55 @@
+#ifndef RA_SERVICES_MOCK_DESKTOP_HH
+#define RA_SERVICES_MOCK_DESKTOP_HH
+#pragma once
+
+#include "services\ServiceLocator.hh"
+
+#include "ui\IDesktop.hh"
+
+namespace ra {
+namespace ui {
+namespace mocks {
+
+using DialogHandler = std::function<ra::ui::DialogResult(WindowViewModelBase&)>;
+
+class MockDesktop : public IDesktop
+{
+public:
+    MockDesktop(DialogHandler pHandler) 
```

Comment:
Should be `explicit `and `noexcept`.
Also `pHandler `should either be passed by `const reference` or `rvalue reference.`

Reply:
ok